Fletcher64.cpp

```diff
@@ -19,25 +18,32 @@ Fletcher64::Fletcher64()
 
 void Fletcher64::begin(uint32_t s1, uint32_t s2)
 {
-  _s1    = s1;
-  _s2    = s2;
+  _s1    = (s1 == FLETCHER_64) ? 0 : s1;
+  _s2    = (s2 == FLETCHER_64) ? 0 : s2;
   _count = 0;
 }
 
 
 void Fletcher64::add(uint32_t value)
 {
   _count++;
+#if defined(ARDUINO_ARCH_AVR)  || defined(ESP32) || defined(ESP8266)
+  uint32_t t = 0xFFFFFFFF - value;
+  if (t >= _s1) _s1 += value;
+  else _s1 = _s1 + value + 1;
+  
+  t = 0xFFFFFFFF - _s1;
+  if (t >= _s2) _s2 += _s1;
+  else _s2 = _s2 + _s1 + 1;
+#elif defined(ARDUINO_ARCH_SAMD)
   _s1 += value;
-#if defined(ARDUINO_ARCH_SAMD) || defined(ESP32) || defined(ESP8266)
   _s1 = (_s1 & ((((uint64_t) 1) << 32) - 1)) + (_s1 >> 32);
-#else
-  if (_s1 >= FLETCHER_64) _s1 -= FLETCHER_64;
-#endif
   _s2 += _s1;
-#if defined(ARDUINO_ARCH_SAMD) || defined(ESP32) || defined(ESP8266)
   _s2 = (_s2 & ((((uint64_t) 1) << 32) - 1)) + (_s2 >> 32);
```

Comment:
`FLETCHER_64` or `((((uint64_t) 1) << 32) - 1)`?

Reply:
FLETCHER64 here as it is more descriptive and much less to read - including counting ()  - for the maintainer.

Reply:
changed number constants to named constants.